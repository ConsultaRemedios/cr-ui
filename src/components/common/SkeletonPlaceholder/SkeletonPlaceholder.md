SkeletonPlaceholder placeholder content component.

```js
<div>
  <SkeletonPlaceholder
    width="18px"
    height="18px"
  />
  <SkeletonPlaceholder
    width="200px"
    height="18px"
  />
  <SkeletonPlaceholder
    width="200px"
    height="18px"
  />
</div>

<div style="margin-left: 20px;">
  <SkeletonPlaceholder
    width="200px"
    height="18px"
  />
</div>
```
