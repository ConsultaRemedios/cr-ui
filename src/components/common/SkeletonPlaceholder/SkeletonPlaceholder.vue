<template>
  <span :class="'cr-ui-skeleton-placeholder'" :style="styles"></span>
</template>

<script>
  export default {
    name: 'SkeletonPlaceholder',

    props: {
      width: {
        type: String,
        default: '',
      },

      height: {
        type: String,
        default: '',
      },
    },

    computed: {
      styles() {
        return `width:${this.width}; height:${this.height}`;
      },
    },
  };
</script>

<style lang="scss">
  @import '../../../styles/components/skeleton-placeholder';
</style>
