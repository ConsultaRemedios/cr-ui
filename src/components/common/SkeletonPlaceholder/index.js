import SkeletonPlaceholder from './SkeletonPlaceholder.vue';

export default SkeletonPlaceholder;
