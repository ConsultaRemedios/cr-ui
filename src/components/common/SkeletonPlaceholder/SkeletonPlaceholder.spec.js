import { shallowMount } from '@vue/test-utils';
import snapshotDiff from 'snapshot-diff';
import SkeletonPlaceholder from './SkeletonPlaceholder.vue';

describe('SkeletonPlaceholder', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(SkeletonPlaceholder);
  });

  it('matches the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('matches the snapshot when props are passed', () => {
    const localWrapper = shallowMount(SkeletonPlaceholder, {
      propsData: {
        width: '20px',
        height: '20px',
      },
    });

    expect(snapshotDiff(wrapper.element, localWrapper.element)).toMatchSnapshot();
  });
});
