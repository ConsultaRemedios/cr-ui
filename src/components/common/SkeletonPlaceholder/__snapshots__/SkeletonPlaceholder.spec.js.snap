// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonPlaceholder matches the snapshot 1`] = `
<span
  class="cr-ui-skeleton-placeholder"
/>
`;

exports[`SkeletonPlaceholder matches the snapshot when props are passed 1`] = `
Snapshot Diff:
- First value
+ Second value

  <span
    class="cr-ui-skeleton-placeholder"
+   style="width: 20px; height: 20px;"
  />
`;
